packages/@aws-cdk/cfnspec/build-tools/split-cfn-to-stack.ts

```diff
@@ -0,0 +1,92 @@
+/* eslint-disable no-console */
+/**
+ * Split the given CloudFormation resource specification up into a patch stack, by service
+ *
+ * After splitting, only write out those fragments that are valid.(*)
+ *
+ * The reason for this is that some service updates sometimes contain mistakes,
+ * and we want to be able to take partial upgrades. By splitting the spec, we can
+ * take updates to some services while leaving updates to other services behind.
+ *
+ * (*) If `NO_VALIDATE` is set, all pieces are written out. In normal operation this
+ *     should not be used.
+ */
+import * as path from 'path';
+import * as fs from 'fs-extra';
+
+import { writeSorted } from './patch-stack';
+import { CfnSpec, CfnSpecValidator, formatErrorInContext } from './validate-cfn';
+
+async function main(args: string[]) {
+  if (args.length < 2) {
+    throw new Error('Usage: split-spec-by-service <SPECFILE> <SCHEMA> <DIRECTORY>');
```

Comment:
`~` I actually like this name `split-spec-by-service` better than `split-cfn-to-stack`.

Usage doesn't appear to match implementation.